tests/e2e/bcd_consumer_integration_test.go

```diff
@@ -1293,17 +1341,19 @@ func (s *BCDConsumerIntegrationTestSuite) registerVerifyConsumer() *bsctypes.Con
 	return registeredConsumer
 }
 
-func (s *BCDConsumerIntegrationTestSuite) finalizeUntilEpoch(epoch uint64) {
+func (s *BCDConsumerIntegrationTestSuite) finalizeUntilEpoch(epoch uint64, wait bool) {
 	bbnClient := s.babylonController.GetBBNClient()
 
 	// wait until the checkpoint of this epoch is sealed
-	s.Eventually(func() bool {
-		lastSealedCkpt, err := bbnClient.LatestEpochFromStatus(ckpttypes.Sealed)
-		if err != nil {
-			return false
-		}
-		return epoch <= lastSealedCkpt.RawCheckpoint.EpochNum
-	}, 1*time.Minute, 1*time.Second)
+	if wait {
```

Comment:
In which situation we should not wait until the given checkpoint is sealed?

Reply:
Every time we need a specific epoch to be finalised.

We just wait for it at the beginning. The rest of the calls, we just finalise the currently sealed epochs.

That said, perhaps there are other ways to write the tests without this. This was the way I found to make the existing tests pass, with "minimal" modifications.